onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -403,7 +400,7 @@ void addGlobalMethods(py::module& m, const Environment& env) {
         std::vector<std::shared_ptr<onnxruntime::IExecutionProviderFactory>> factories = {
             onnxruntime::CreateExecutionProviderFactory_CPU(0),
 #ifdef USE_CUDA
-            onnxruntime::CreateExecutionProviderFactory_CUDA(0),
+            onnxruntime::CreateExecutionProviderFactory_CUDA(cuda_device_id, cuda_mem_limit, arena_extend_strategy),
 #endif
```

Comment:
This change probably isn't necessary as all the EPs here are created with default args as the instances are only used to get the kernel defs.

Reply:
This was the core change in this PR. This was causing the build break. I think previously some arguments had default values. Now, we need to explicitly provide them.

Reply:
Oh, the training PR changed the signature. But we'll never be able to do the corresponding change on the C API, even though the  CreateExecutionProviderFactory_CUDA function was created for the C APIs.
We should support multiple GPU devices. Hector Li has a test pipeline for  that feature and it runs daily. But as you can see, as long as we hard code the '0' here. we can't really support multiple GPUs through the python interface.

Reply:
yes, the training PR changed the signature and this change basically accommodates that.  I would think that for the the doc gen exercise, the intricacies involved with different GPU devices should not matter given that we only create the CUDA EP to query its kernel registry for the sake of creating the documentation (please correct me if I am wrong). And you are right, here the change is equivalent to hard-coding '0' for the device id.